experimental/visual-dataflow/src/DOTParser.cpp

```diff
@@ -0,0 +1,184 @@
+#include "DOTParser.h"
+#include "DOTReformat.h"
+#include "Graph.h"
+#include "GraphEdge.h"
+#include "GraphNode.h"
+#include "mlir/Support/LogicalResult.h"
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <set>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using namespace dynamatic::experimental::visual_dataflow;
+
+LogicalResult DOTParser::processDOT(std::ifstream &file, Graph *graph) {
+
+  std::string line;
+
+  EdgeId currentEdgeID = 0;
+  GraphNode *currentNode;
+  bool insideNodeDefinition = false;
+  GraphEdge *currentEdge;
+  bool insideEdgeDefinition = false;
+
+  std::regex positionRegex("e,\\s*([0-9]+),([0-9]+(\\.[0-9]+)?)");
+  std::smatch match;
+
+  while (std::getline(file, line)) {
+    if (line.find("->") == std::string::npos &&
+        line.find('[') != std::string::npos &&
+        line.find("node") == std::string::npos &&
+        line.find("graph") == std::string::npos) {
+
+      currentNode = new GraphNode();
+      insideNodeDefinition = true;
+    }
+
+    else if (line.find("->") != std::string::npos) {
```

Comment:
nit: it's conventional to put the `else` on the same line as the `if`'s closing bracket, like
```cpp
} else if (...) {
```